llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -535,8 +535,7 @@ createScalarIVSteps(VPlan &Plan, InductionDescriptor::InductionKind Kind,
   }
 
   // Truncate base induction if needed.
-  VPTypeAnalysis TypeInfo(Plan.getCanonicalIV()->getScalarType(),
-                          SE.getContext());
+  VPTypeAnalysis TypeInfo(Plan.getCanonicalIV()->getScalarType(), Ctx);
```

Comment:
Would it be safe to use `Plan.getCanonicalIV()->getScalarType()->getContext` here as done by `willGenerateVectors`?

Reply:
Good point! I've spotted some more opportunities for clean up here. For example, the class constructor for `VPTypeAnalysis` takes both a Type and a LLVMContext, when you only need the former. I'll follow that up in another patch. :)

Reply:
Yep that should be safe